spanner/client_test.go

```diff
@@ -591,6 +591,62 @@ func testReadOnlyTransaction(t *testing.T, executionTimes map[string]SimulatedEx
 	return executeSingerQuery(ctx, tx)
 }
 
+func TestClient_ReadWriteTransaction_StatementBased(t *testing.T) {
+	t.Parallel()
+
+	rowCount, err := testReadWriteTransactionStatementBased(t, make(map[string]SimulatedExecutionTime))
+	if err != nil {
+		t.Fatalf("transaction failed to commit: %v", err)
+	}
+	if rowCount != SelectSingerIDAlbumIDAlbumTitleFromAlbumsRowCount {
+		t.Fatalf("Row count mismatch, got %v, expected %v", rowCount, SelectSingerIDAlbumIDAlbumTitleFromAlbumsRowCount)
+	}
+}
+
+func TestClient_ReadWriteTransaction_StatementBased_CommitAborted(t *testing.T) {
+	t.Parallel()
+	_, err := testReadWriteTransactionStatementBased(t, map[string]SimulatedExecutionTime{
+		MethodCommitTransaction: {Errors: []error{status.Error(codes.Aborted, "Transaction aborted")}},
+	})
+	if status.Code(err) != codes.Aborted || !strings.Contains(err.Error(), "Transaction aborted") {
+		t.Fatalf("got an unexpected error: %v", err)
+	}
+}
+
+func testReadWriteTransactionStatementBased(t *testing.T, executionTimes map[string]SimulatedExecutionTime) (int64, error) {
+	server, client, teardown := setupMockedTestServer(t)
+	defer teardown()
+	for method, exec := range executionTimes {
+		server.TestSpanner.PutExecutionTime(method, exec)
+	}
+	ctx := context.Background()
+
+	tx, err := client.BeginReadWriteTransaction(ctx)
```

Comment:
I would suggest that we let this test case implement a retry loop for `Aborted` errors. That would prevent users who look at these test cases for an example on how to use a particular feature from assuming that a retry-loop is not needed. The tests that check that an `Aborted` error is returned and handled accordingly should check for a certain number of retry attempts, instead of checking for the existence of an `Aborted` error return value.